test/csuite/parse_opts/main.c

```diff
@@ -0,0 +1,315 @@
+ * successors. We intend this dedication to be an overt act of
+ * relinquishment in perpetuity of all present and future rights to this
+ * software under copyright law.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+ * OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+#include "test_util.h"
+
+/*
+ * Unit test testutil_parse_opts and testutil_parse_opt functions.
+ */
+
+extern char *__wt_optarg; /* option argument */
+extern int __wt_optind;   /* argv position, needed to reset __wt_getopt parser */
+extern int __wt_optreset; /* needed to reset the parser internal state */
+
+/*
+ * This structure aids in testing testutil_parse_opt. That function is useful for test applications
+ * that wish to extend/modify the basic option set provided by testutil.
+ */
+typedef struct {
+    char *c_option;
+    bool d_option;
+    bool e_option;
+    int f_option;
+} EXTENDED_OPTS;
+
+/*
+ * This drives the testing. For each given command_line, there is a matching expected TEST_OPTS
+ * structure. When argv[0] is "parse_opts", we are driving testutil_parse_opts. If argv[0] is
+ * "parse_opt", then we are parsing some of our "own" options, put into an EXTENDED_OPTS struct, and
+ * using testutil_parse_opt to parse any we don't recognize, those are put into TEST_OPTS.
+ */
+static const char *command_line0[] = {"parse_opts", "-b", "builddir", "-T", "21", NULL};
+static const char *command_line1[] = {"parse_opts", "-bbuilddir", "-T", "21", NULL};
+static const char *command_line2[] = {"parse_opts", "-v", "-PT", NULL};
+static const char *command_line3[] = {"parse_opts", "-v", "-Po", "my_store", "-PT", NULL};
+static const char *command_line4[] = {"parse_opts", "-v", "-Pomy_store", "-PT", NULL};
+
+static const char *command_line5[] = {"parse_opt", "-vd", "-Pomy_store",
+  "-c"
+  "string_opt",
+  "-PT", NULL};
+static const char *command_line6[] = {
+  "parse_opt", "-dv", "-Pomy_store", "-cstring_opt", "-PT", NULL};
+static const char *command_line7[] = {
+  "parse_opt", "-ev", "-cstring_opt", "-Pomy_store", "-PT", "-f", "22", NULL};
+static const char *command_line8[] = {"parse_opt", "-evd", "-Pomy_store", "-PT", "-f22", NULL};
+static const char *command_line9[] = {"parse_opt", "-v", "-Pomy_store", "-PT", NULL};
+
+static const char **command_lines[] = {
+  command_line0,
+  command_line1,
+  command_line2,
+  command_line3,
+  command_line4,
+  command_line5,
+  command_line6,
+  command_line7,
+  command_line8,
+  command_line9,
+};
+
+static TEST_OPTS expected[] = {
+  {NULL, NULL, {0}, NULL, (char *)"builddir", NULL, 0, NULL, NULL, false, false, false, false, 0, 0,
```

Comment:
I might suggest defining these together and in the same way. So maybe something like this:
static const char *command_line0[] = {"parse_opts", "-b", "builddir", "-T", "21", NULL};
static TEST_OPTS expected0 = {NULL, NULL, {0}, NULL, (char *)"builddir", NULL, 0, NULL, NULL, false, false, false, false, 0, 0, 21, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0};

static TEST_OPTS expected[] = { expected0, ... };

So that someone adding another one in the future would see they need to add them together.

Reply:
I reworked this to keep the relevant parts together and remove the extra symbols, so I think it's a lot clearer.